paddle/framework/op_info.h

```diff
@@ -25,16 +25,10 @@
 namespace paddle {
 namespace framework {
 
-class GradOpDescMakerBase {
- public:
-  virtual ~GradOpDescMakerBase() = default;
-  virtual std::vector<OpDescBind> operator()(const OpDescBind&) const = 0;
-};
-
 struct OpInfo {
   OpCreator creator_;
   std::string grad_op_type_;
-  GradOpDescMakerBase* grad_op_maker_{nullptr};
+  std::function<std::vector<OpDescBind>(const OpDescBind&)> grad_op_maker_;
```

Comment:
`grad_op_maker_` returns `std::vector<OpDescBind>`, while in `BlockDescBind` `ops_` is `std::deque<std::unique_ptr<OpDescBind>>`. So when and how shall we convert `OpDescBind` to `std::unique_ptr<OpDescBind>`?

Reply:
Yes. You are right. `grad_op_maker_` should return `vector<unique_ptr<OpDescBind>>`.